cmd/log_index.cpp

```diff
@@ -0,0 +1,241 @@
+#include <silkworm/db/bitmap.hpp>
+#include <silkworm/db/stages.hpp>
+#include <silkworm/db/tables.hpp>
+#include <silkworm/etl/collector.hpp>
+#include <cbor/decoder.h>
+
+using namespace silkworm;
+
+constexpr size_t kBitmapBufferSizeLimit = 512 * kMebi;
+
+
+void loader_function(etl::Entry entry, lmdb::Table *target_table, unsigned int db_flags) {
+    auto bm{roaring::Roaring::readSafe(byte_ptr_cast(entry.value.data()), entry.value.size())};
+    Bytes last_chunk_index(entry.key.size() + 4, '\0');
+    std::memcpy(&last_chunk_index[0], &entry.key[0], entry.key.size());
+    boost::endian::store_big_u32(&last_chunk_index[entry.key.size()], UINT32_MAX);
+    auto previous_bitmap_bytes{target_table->get(last_chunk_index)};
+    if (previous_bitmap_bytes.has_value()) {
+        bm |= roaring::Roaring::readSafe(byte_ptr_cast(previous_bitmap_bytes->data()),
+                                                previous_bitmap_bytes->size());
+        db_flags = 0;
+    }
+    while (bm.cardinality() > 0) {
+        auto current_chunk{db::bitmap::cut_left(bm, db::bitmap::kBitmapChunkLimit)};
+        // make chunk index
+        Bytes chunk_index(entry.key.size() + 4, '\0');
+        std::memcpy(&chunk_index[0], &entry.key[0], entry.key.size());
+        uint64_t suffix{bm.cardinality() == 0 ? UINT32_MAX : current_chunk.maximum()};
+        boost::endian::store_big_u32(&chunk_index[entry.key.size()], suffix);
+        Bytes current_chunk_bytes(current_chunk.getSizeInBytes(), '\0');
+        current_chunk.write(byte_ptr_cast(&current_chunk_bytes[0]));
+        target_table->put(chunk_index, current_chunk_bytes, db_flags);
+    }
+}
+
+class listener_log_index : public cbor::listener {
+    public:
+    listener_log_index(uint64_t block_number, std::unordered_map<std::string, roaring::Roaring> * topics_map,
+                std::unordered_map<std::string, roaring::Roaring> * addrs_map, uint64_t * allocated_topics, uint64_t * allocated_addrs_): 
+                block_number_(block_number), topics_map_(topics_map), addrs_map_(addrs_map), allocated_topics_(allocated_topics), allocated_addrs_(allocated_addrs_) {};
+
+    virtual void on_integer(int){};
+
+    virtual void on_bytes(unsigned char *data, int size) {
+        std::string key(reinterpret_cast<const char *>(data), size);
+        if (size == kHashLength) {
+            if (topics_map_->find(key) == topics_map_->end()) {
+                topics_map_->emplace(key, roaring::Roaring());
+            }
+            topics_map_->at(key).add(block_number_);
+            *allocated_topics_ += kHashLength;
+        } else if (size == kAddressLength) {
+            if (addrs_map_->find(key) == addrs_map_->end()) {
+                addrs_map_->emplace(key, roaring::Roaring());
+            }
+            addrs_map_->at(key).add(block_number_);
+            *allocated_addrs_ += kAddressLength;
+        }
+        delete[] data;
+    }
+
+    virtual void on_string(std::string &) {};
```

Comment:
```
void on_string(std::string &) override {};
```
instead of
```
virtual void on_string(std::string &) {};
```

Rational: [Google's Style Guide](https://google.github.io/styleguide/cppguide.html#Inheritance)